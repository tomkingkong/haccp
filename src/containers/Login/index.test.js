import React from 'react';
import { shallow } from 'enzyme';

import { Signup } from '.';
import { jssPreset } from '@material-ui/core';

describe('Signup Component', () => {
let wrapper;

  it('should match snapshot', () => {
    wrapper = shallow(<Signup />);
    expect(wrapper).toMatchSnapshot();
	});
	
	it('should handle a change', () => {
		wrapper = shallow(<Signup />);
		const mockEvent = {
			target: {
				name: 'email',
				value: 'ben@gmail.com'
			}
		};
		expect(wrapper.state('email')).toEqual('');
		wrapper.instance().handleChange(mockEvent);
		expect(wrapper.state('email')).toEqual('ben@gmail.com');

	});

	xit('should handle submit', () => {
		window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
			json: () => Promise.resolve({id:1})
			})
		);
		
		const mockHistory = [];
		wrapper = shallow(<Signup history={mockHistory}/>);
		wrapper.setState({
			email: 'email@gmail.com',
			password: 'password',
			name: 'ben'});

		const mockEvent = {
			preventDefault: jest.fn()
		};
		wrapper.instance().handleSubmit(mockEvent);
		expect(mockHistory).toEqual(['/companyinfo']);
	});
});