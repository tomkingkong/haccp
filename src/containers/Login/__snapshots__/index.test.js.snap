// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should match snapshot 1`] = `
<div>
  <h2>
    Login
  </h2>
  <form
    onSubmit={[Function]}
  >
    <input
      name="email"
      onChange={[Function]}
      value=""
    />
    <input
      name="password"
      onChange={[Function]}
      value=""
    />
    <button>
      Log In
    </button>
  </form>
</div>
`;
